### PR TITLE
feat: clean up report service

### PR DESCRIPTION
**Pull request by bistaastha**

### CODE REVIEW

**Review by mvaishnavi — APPROVED** [review]

File: src/app/core/services/report.service.ts [hunk 1]
```diff
@@ -438,27 +263,6 @@ export class ReportService {
...
-          queryParams: config.queryParams,
-          order: config.order,
-        })
-      ),
-      map((res) => res.data),
-      reduce((acc, curr) => acc.concat(curr), [] as ExtendedReport[])
-    );
-  }
-
   addOrderByParams(
```
Comment (mvaishnavi) [inline src/app/core/services/report.service.ts:266]: is `searchParamsGenerator` still being used?
can we get rid of it if not along with this method?

Comment (mvaishnavi) [reply src/app/core/services/report.service.ts:266]: similar check and removal needed for 
`getApproversInBulk`
`userReportsSearchParamsGenerator`
`addApprovers`

**Review by mvaishnavi — APPROVED** [review]
minor comments; remove all unused methods test well and merge in.

File: src/app/core/services/report.service.ts (hunk 1 above)
Comment (bistaastha) [reply src/app/core/services/report.service.ts]: Removed these methods

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**